OpenAI.Images.pas

```diff
@@ -409,46 +439,90 @@ function TImageSizeHelper.ToString: string;
 
 { TAzureImageResponse }
 
-constructor TAzureImageResponse.Create;
-begin
-  FResult := TAzureImageData.Create;
-end;
-
 destructor TAzureImageResponse.Destroy;
 begin
-  FResult.Free;
+  if Assigned(FResult) then
+    FResult.Free;
+  if Assigned(FError) then
+    FError.Free;
   inherited;
 end;
 
 { TImagesAzureRoute }
 
 function TImagesAzureRoute.Create(ParamProc: TProc<TImageAzureCreateParams>): TAzureImageResponse;
 var
-  ARequest: TAzureImageRequest;
+  StartTime, ElapsedTime: UInt64;
+const
+  Timeout: Integer = 20000; // 20 second timeout
+  PollInterval: Integer = 1000; // poll for image once per second
 begin
   // First place the task with POST
-  ARequest := API.Post<TAzureImageRequest, TImageAzureCreateParams>('text-to-image', ParamProc);
+  Result := API.Post<TAzureImageResponse, TImageAzureCreateParams>('text-to-image', ParamProc);
+
+  // Check if we got a valid id - current azure documentation is not that precise here if we always get "NotStarted".
+  // Otherwise we get "failed" and in the "error"-property we can find "code" and "message" of the error
+  if (Result.ID = '') or (Result.Status = 'Failed') then
+    exit;
+
+  StartTime := GetTickCount64;
```

Comment:
This should be replaced with TThread.GetTickCount to remove platform dependency

Reply:
Wow - you are fast ;)